src/erlcloud_s3.erl

```diff
@@ -1141,9 +1154,37 @@ aws_region_from_host(Host) ->
     case string:tokens(Host, ".") of
         ["s3", _, _] ->
             "us-east-1";
+        ["s3", Value, _, _] ->
+            Value;
         [Value, _, _] ->
             % Skip "s3-" prefix
             string:substr(Value, 4);
         _ ->
             "us-east-1"
     end.
+
+%%
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/Redirects.html
+%%
+s3_follow_redirect(Response, Config, Method, Bucket, Path, Subresource, Params, POSTData, Headers) ->
+    case Response of
+        %% Handle redirect responce once.
+        {error, {http_error, StatusCode, _, ErrBody, ErrHeaders}} = _Err
+            when StatusCode >= 300 andalso StatusCode < 400 ->
```

Comment:
I'll exclude 300 (MultipleChoises). The other 3xx (except 304) responses should normally contain location header.